internal/gatewayapi/translator.go

```diff
@@ -546,11 +548,129 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
 						}
 					}
 
+					// TODO: implement header modifier filter
+					for _, filter := range rule.Filters {
```

Comment:
It might make sense to move this out of the loop on `rule.Matches` since the same set of filters apply for all matches in a given Gateway API route rule, that way we can avoid processing the filters multiple times, once per match. But, not a big deal right now and we can always revisit.